File: src/frontend/src/app/core/models/learning-outcome/learning-outcome.ts
import { LearningOutcomeCategory } from "../../enums/learning-outcome/learning-outcome-category.enum";
import { Specialization } from "../field-of-study/specialization";
import { ModelBase } from "../model-base";

export class LearningOutcome extends ModelBase {
    symbol: string = '';
    category: LearningOutcomeCategory = LearningOutcomeCategory.Knowledge;
    description?: string;
    u1degreeCharacteristics: string = '';
    s2degreePrk?: string;
    s2degreePrkeng?: string;
    specialization?: Specialization;
}

File: src/backend/SyllabusManager.Data/Models/LearningOutcomes/LearningOutcome.cs
using SyllabusManager.Data.Attributes;
using SyllabusManager.Data.Enums.LearningOutcomes;
using SyllabusManager.Data.Models.FieldOfStudies;
using System.ComponentModel.DataAnnotations;

namespace SyllabusManager.Data.Models.LearningOutcomes
{
    public class LearningOutcome : ModelBase
    {
        [Required]
        [PdfName("Symbol kierunkowych efektów uczenia się")]
        public string Symbol { get; set; }
        public LearningOutcomeCategory Category { get; set; }
        [PdfName("Opis efektów uczenia się dla kierunku studiów")]
        public string Description { get; set; }
        [Required]
        [PdfName("Uniwersalne charakterystyki pierwszego stopnia(U)")]
        public string U1degreeCharacteristics { get; set; }
        [PdfName("Charakterystyki dla kwalifikacji na poziomie 6 PRK")]
        public string S2degreePrk { get; set; }
        [PdfName("Charakterystyki dla kwalifikacji na poziomie 6 PRK, umożliwiających uzyskanie kompetencji inżynierskich")]
        public string S2degreePrkeng { get; set; }

        public Specialization Specialization { get; set; }
    }
}
